client/repoutils.c

```diff
@@ -680,3 +680,101 @@ TDNFRepoApplyProxySettings(
 error:
     goto cleanup;
 }
+
+uint32_t
+TDNFRepoApplySSLSettings(
+    PTDNF pTdnf,
+    const char* pszRepo,
+    CURL *pCurl
+    )
+{
+    uint32_t dwError = 0;
+    PTDNF_REPO_DATA_INTERNAL pRepos = NULL;
+
+    if(!pTdnf || IsNullOrEmptyString(pszRepo) || !pCurl)
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+    if(!pTdnf->pRepos)
+    {
+        dwError = ERROR_TDNF_NO_REPOS;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+    pRepos = pTdnf->pRepos;
+
+    while(pRepos)
+    {
+        if(!strcmp(pszRepo, pRepos->pszId))
+        {
+            break;
+        }
+        pRepos = pRepos->pNext;
+    }
+
+    if(!pRepos)
+    {
+        dwError = ERROR_TDNF_REPO_NOT_FOUND;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    if(curl_easy_setopt(
+            pCurl,
+            CURLOPT_SSL_VERIFYPEER,
+            ((pRepos->nSSLVerify) ? 1 : 0)) != CURLE_OK)
+    {
+        dwError = ERROR_TDNF_SET_SSL_SETTINGS;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
```

Comment:
Can we get a more detailed error here from curl? Same for the other 3 `curl_easy_setopt()` calls.

Reply:
Pls refer this for setting curl error

https://github.com/vmware/tdnf/blob/31a577946b6286daf10c7af5f1b94f36a8b2e59f/client/remoterepo.c#L793

Reply:
I tried to replicated the behavior from `TDNFRepoApplyProxySettings` earlier in this file. I frankly feel like pointing out which user input caused the error is more useful than the `curl`'s internal error. In most of the cases I have encountered, `curl` already prints to stderr about what went wrong anyway.

Reply:
So if curl prints out to stderr before returning this might be good enough - we just need information what went wrong. Can you give an example?

Reply:
Here is a sample:

```
curl#60: Peer certificate cannot be authenticated with given CA certificates
```